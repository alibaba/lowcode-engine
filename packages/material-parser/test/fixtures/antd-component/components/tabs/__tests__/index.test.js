import React from 'react';
import { mount, render } from 'enzyme';
import Tabs from '..';
import mountTest from '../../../tests/shared/mountTest';
import rtlTest from '../../../tests/shared/rtlTest';

const { TabPane } = Tabs;

describe('Tabs', () => {
  mountTest(() => (
    <Tabs>
      <TabPane tab="xx" key="xx" />
    </Tabs>
  ));
  rtlTest(() => (
    <Tabs>
      <TabPane tab="xx" key="xx" />
    </Tabs>
  ));

  describe('editable-card', () => {
    let handleEdit;
    let wrapper;

    beforeEach(() => {
      handleEdit = jest.fn();
      wrapper = mount(
        <Tabs type="editable-card" onEdit={handleEdit}>
          <TabPane tab="foo" key="1">
            foo
          </TabPane>
          {undefined}
          {null}
          {false}
        </Tabs>,
      );
    });

    it('add card', () => {
      wrapper
        .find('.ant-tabs-new-tab')
        .hostNodes()
        .simulate('click');
      expect(handleEdit.mock.calls[0][1]).toBe('add');
    });

    it('remove card', () => {
      wrapper.find('.anticon-close').simulate('click');
      expect(handleEdit).toHaveBeenCalledWith('1', 'remove');
    });

    it('validateElement', () => {
      expect(wrapper.find('.ant-tabs-tab').length).toBe(1);
    });
  });

  describe('tabPosition', () => {
    it('remove card', () => {
      const wrapper = render(
        <Tabs tabPosition="left" tabBarExtraContent="xxx">
          <TabPane tab="foo" key="1">
            foo
          </TabPane>
        </Tabs>,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('renderTabBar', () => {
    it('custom-tab-bar', () => {
      const wrapper = render(
        <Tabs renderTabBar={() => <div>custom-tab-bar</div>}>
          <TabPane tab="foo" key="1">
            foo
          </TabPane>
        </Tabs>,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
