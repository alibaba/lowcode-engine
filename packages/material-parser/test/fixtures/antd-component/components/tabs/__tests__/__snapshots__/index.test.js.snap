// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renderTabBar custom-tab-bar 1`] = `
<div
  class="ant-tabs ant-tabs-top ant-tabs-line"
>
  <div>
    custom-tab-bar
  </div>
  <div
    class="ant-tabs-content ant-tabs-content-animated ant-tabs-top-content"
    style="margin-left:0%"
  >
    <div
      aria-hidden="false"
      class="ant-tabs-tabpane ant-tabs-tabpane-active"
      role="tabpanel"
      style="visibility:visible"
      tabindex="0"
    >
      foo
    </div>
  </div>
</div>
`;

exports[`Tabs rtl render component should be rendered correctly in RTL direction 1`] = `
<div
  class="ant-tabs ant-tabs-top ant-tabs-line ant-tabs-rtl"
>
  <div
    class="ant-tabs-bar ant-tabs-top-bar"
    role="tablist"
    tabindex="-1"
  >
    <div
      class="ant-tabs-nav-container"
    >
      <span
        class="ant-tabs-tab-prev ant-tabs-tab-btn-disabled"
        unselectable="unselectable"
      >
        <span
          class="ant-tabs-tab-prev-icon"
        >
          <span
            aria-label="right"
            class="anticon anticon-right ant-tabs-tab-prev-icon-target"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        class="ant-tabs-tab-next ant-tabs-tab-btn-disabled"
        unselectable="unselectable"
      >
        <span
          class="ant-tabs-tab-next-icon"
        >
          <span
            aria-label="left"
            class="anticon anticon-left ant-tabs-tab-next-icon-target"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="left"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
              />
            </svg>
          </span>
        </span>
      </span>
      <div
        class="ant-tabs-nav-wrap"
      >
        <div
          class="ant-tabs-nav-scroll"
        >
          <div
            class="ant-tabs-nav ant-tabs-nav-animated"
          >
            <div>
              <div
                aria-controls="tabpane-xx"
                aria-disabled="false"
                aria-selected="true"
                class="ant-tabs-tab-active ant-tabs-tab"
                id="tab-xx"
                role="tab"
                tabindex="0"
              >
                xx
              </div>
            </div>
            <div
              class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-tabs-content ant-tabs-content-animated ant-tabs-top-content"
    style="margin-left:100%"
  >
    <div
      aria-hidden="false"
      class="ant-tabs-tabpane ant-tabs-tabpane-active"
      role="tabpanel"
      style="visibility:visible"
      tabindex="0"
    />
  </div>
</div>
`;

exports[`Tabs tabPosition remove card 1`] = `
<div
  class="ant-tabs ant-tabs-left ant-tabs-vertical ant-tabs-line"
>
  <div
    class="ant-tabs-bar ant-tabs-left-bar"
    role="tablist"
    tabindex="-1"
  >
    <div
      class="ant-tabs-nav-container"
    >
      <span
        class="ant-tabs-tab-prev ant-tabs-tab-btn-disabled"
        unselectable="unselectable"
      >
        <span
          class="ant-tabs-tab-prev-icon"
        >
          <span
            aria-label="up"
            class="anticon anticon-up ant-tabs-tab-prev-icon-target"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="up"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M890.5 755.3L537.9 269.2c-12.8-17.6-39-17.6-51.7 0L133.5 755.3A8 8 0 00140 768h75c5.1 0 9.9-2.5 12.9-6.6L512 369.8l284.1 391.6c3 4.1 7.8 6.6 12.9 6.6h75c6.5 0 10.3-7.4 6.5-12.7z"
              />
            </svg>
          </span>
        </span>
      </span>
      <span
        class="ant-tabs-tab-next ant-tabs-tab-btn-disabled"
        unselectable="unselectable"
      >
        <span
          class="ant-tabs-tab-next-icon"
        >
          <span
            aria-label="down"
            class="anticon anticon-down ant-tabs-tab-next-icon-target"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="down"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
              />
            </svg>
          </span>
        </span>
      </span>
      <div
        class="ant-tabs-nav-wrap"
      >
        <div
          class="ant-tabs-nav-scroll"
        >
          <div
            class="ant-tabs-nav ant-tabs-nav-animated"
          >
            <div>
              <div
                aria-controls="tabpane-1"
                aria-disabled="false"
                aria-selected="true"
                class="ant-tabs-tab-active ant-tabs-tab"
                id="tab-1"
                role="tab"
                tabindex="0"
              >
                foo
              </div>
            </div>
            <div
              class="ant-tabs-ink-bar ant-tabs-ink-bar-animated"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-tabs-extra-content"
    >
      xxx
    </div>
  </div>
  <div
    class="ant-tabs-content ant-tabs-content-animated ant-tabs-left-content"
    style="margin-top:0%"
  >
    <div
      aria-hidden="false"
      class="ant-tabs-tabpane ant-tabs-tabpane-active"
      role="tabpanel"
      style="visibility:visible"
      tabindex="0"
    >
      foo
    </div>
  </div>
</div>
`;
